pkg/controller/master/backup/backup.go

```diff
@@ -714,8 +714,9 @@ func (h *Handler) configureBackupTargetOnStatus(vmBackup *harvesterv1.VirtualMac
 		BucketName:   vmBackup.Annotations[backupBucketNameAnnotation],
 		BucketRegion: vmBackup.Annotations[backupBucketRegionAnnotation],
 	}
-	delete(vmBackup.Annotations, backupTargetAnnotation)
-	delete(vmBackup.Annotations, backupBucketNameAnnotation)
-	delete(vmBackup.Annotations, backupBucketRegionAnnotation)
-	return h.vmBackups.Update(vmBackupCpy)
+	delete(vmBackupCpy.Annotations, backupTargetAnnotation)
+	delete(vmBackupCpy.Annotations, backupBucketNameAnnotation)
+	delete(vmBackupCpy.Annotations, backupBucketRegionAnnotation)
+	_, err := h.vmBackups.Update(vmBackupCpy)
```

Comment:
Would prefer to update the status directly and don't need to delete the annotations, also we can skip updating when the `status.backupTarget` is not empty, WDUT?

Reply:
I tried it, but it looks like we don't have `UpdateStatus` in `VirtualMachineBackupClient`.

https://github.com/harvester/harvester/blob/655c251ada5808399c8b9c475abc75e437a51d09/pkg/generated/controllers/harvesterhci.io/v1beta1/virtualmachinebackup.go#L55-L64

Reply:
You can get a generated `UpdateStatus` by dropping the pointer and using the struct directly.

https://github.com/harvester/harvester/blob/655c251ada5808399c8b9c475abc75e437a51d09/pkg/apis/harvesterhci.io/v1beta1/backup.go#L50

Reply:
> 

@guangbochen regarding your question `"also we can skip updating when the status.backupTarget is not empty"`

This commit check `isBackupMissingStatus`, normally when it is resource created before v1.0.0, it is true, then create a `backupCpy.Status` pointer, a `backupCpy.Status.BackupTarget` pointer,  they are considered as place holder to differentiate old and new, also did the upgrade work. Finally update self.

no matter the value of the old backup target, the update of self looks is essential.


```
func isBackupMissingStatus(backup *harvesterv1.VirtualMachineBackup) bool {
	return backup.Status == nil || backup.Status.SourceSpec == nil || backup.Status.VolumeBackups == nil || backup.Status.BackupTarget == nil
}
```

```
func (h *Handler) OnBackupChange(key string, vmBackup *harvesterv1.VirtualMachineBackup) (*harvesterv1.VirtualMachineBackup, error) {
if isBackupMissingStatus(vmBackup) {
..
		return nil, h.initBackup(vmBackup, sourceVM, target)
..
}
```
	
```
func (h *Handler) initBackup(backup *harvesterv1.VirtualMachineBackup, vm *kv1.VirtualMachine, target *settings.BackupTarget) error {
...
backupCpy.Status = &harvesterv1.VirtualMachineBackupStatus{

	backupCpy.Status.BackupTarget = &harvesterv1.BackupTarget{
		Endpoint:     target.Endpoint,
		BucketName:   target.BucketName,
		BucketRegion: target.BucketRegion,
	}

	if _, err := h.vmBackups.Update(backupCpy); err != nil {
		return err
	}
	return nil
}
```

Reply:
> a `backupCpy.Status.BackupTarget` pointer, they are considered as place holder to differentiate old and new, also did the upgrade work. Finally update self.

@weihanglo , I am not sure whether we can backport from pointer to struct. If we want to use struct, I will suggest let's do it in the next version API like `v1beta2`.